### PR TITLE
feat: feegrant module for blobs submission via DA Light node

### LINKED ISSUE #1470: feat: fee grant module doc
```[tasklist]
### Tasks
- [x] add and test celestia-app portion of guide
- [ ] add and test celestia-node portion of guide
```

### PR DESCRIPTION
**Pull request by Bidon15**
## Overview
Works: 

- https://mocha.celenium.io/address/celestia1rgj3z5rfydj3xulhm6glqr0zs7j8rr9rfh6ekw
- https://mocha.celenium.io/tx/f28aeffeb003a21d4810660bb7d529b744d24419440e29080b3284a247977fe1
- https://mocha.celenium.io/tx/802a17777fbeab416f6fa2c25f0c56dd9cc8a92afc2a96293d114ac7c22efb5c

Resolves #1470 

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

## Summary by CodeRabbit

- **New Features**
	- Introduced a guide on using the FeeGrant module for blobs submission on Celestia's Mocha testnet, allowing third-party accounts to cover transaction fees.

### CODE REVIEW

File: developers/feegrant-for-blobs.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+## Granting fee allowances
+
+To grant fee allowances, allowing a third-party (granter) account to pay
+for the fees incurred by a Celestia data availability node (grantee)
+account, use the following command:
+
+```bash
+celestia-appd tx feegrant grant \
+  [grantee_address] [granter_address] \
+  --home [path_to_grantee] \
```
Comment (jcstein) [inline developers/feegrant-for-blobs.md:37]: should this be `[path_to_granter]` @Bidon15 ?

Comment (jcstein) [reply developers/feegrant-for-blobs.md:36]: I cannot get it to work with:

```bash
celestia-appd tx feegrant grant \
  celestia18uguxd6avku8zrcugmlvjcfnm7wuc53ej2ujhn celestia1spgm03y9ctpnw9lzqpuxrf6qvwszrqtnq4sjs7 \
  --home ~/.celestia-light-mocha-4/ \
  --node https://rpc.celestia-mocha.com:443 \
  --spend-limit 1000000utia \
  --allowed-messages "/cosmos.bank.v1beta1.MsgSend,/celestia.blob.v1.MsgPayForBlobs" \
  --chain-id mocha-4 \
  --keyring-backend test \
  --fees 20000utia \
  --broadcast-mode block \
  --yes
Error: key with address celestia18uguxd6avku8zrcugmlvjcfnm7wuc53ej2ujhn not found: key not found
```

Where `celestia18uguxd6avku8zrcugmlvjcfnm7wuc53ej2ujhn` is grantee (light node account) and `celestia1spgm03y9ctpnw9lzqpuxrf6qvwszrqtnq4sjs7` is the celestia-app account. I've tried different paths:
1. `--home ~/.celestia-light-mocha-4/`
2. `--home ~/.celestia-light-mocha-4/data/keys/`
3. `--home ~/.celestia-light-mocha-4/data/keys/keyring-test/josh.info`
4. `--home ~/.celestia-light-mocha-4/data/keys/keyring-test/`

File: developers/feegrant-for-blobs.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
+# FeeGrant module for blobs submission
+
+## Overview
+
+This guide provides developers with the knowledge to use the FeeGrant
+module on the Celestia's Mocha testnet chain for granting a data
+availability node's account to submit blobs without constantly
+funding it, enabling a third-party account to cover the transaction fees.
```
[the remark this answers is not in the document]
Comment (jcstein) [reply developers/feegrant-for-blobs.md:8]: 99% sure this is incorrect, so keeping it with a hyphen

it's not a third party-account, or a third party account, but a third-party account

**Comment by jcstein** [comment]
This PR also covers a basic example of #1492

**Comment by jcstein** [comment]
notes from @vgonkivs:

I’ve just merged the fee granting pr. Some TLDR for you for docs updating:
- using node you now can easily give permission for other nodes to submit transactions on your behalf(I mean your node will pay fees for these txs)
- you can revoke the grant as well
The node that receives the grant has to run a node with  --`granter.address=celestia1v96seg23ehfhfjwk0wcy3s5y0krptt284vh236` to use granting functionality. The granter address will be stored until the next run of your local node. So, in case the granter revokes permission, you will have to restart the node without this flag. I guess, important to know that such transactions(with granter) will consume more gas than the regular ones.
https://github.com/celestiaorg/celestia-node/pull/3304 - here is the PR.
to grant permission, you’ll have to execute:
```bash
celestia state grant-fee celestia12psg90lwvxdktnqerr2p5mxuw2v3je497uv6tj [2000 1000000](tel:20001000000) --node.store ~/.celestia-light-mocha-4/
```
to revoke:
```bash
celestia state revoke-grant-fee celestia12psg90lwvxdktnqerr2p5mxuw2v3je497uv6tj [2000 1000000](tel:20001000000) --node.store ~/.celestia-light-mocha-4/
```
to run the node in a grantee mode:
```bash
celestia full start --core.ip full.consensus.mocha-4.celestia-mocha.com --p2p.network=mocha --granter.address=celestia1v96seg23ehfhfjwk0wcy3s5y0krptt284vh236
```

This PR also added a gasMultiplier, for the transactions with the Options. Not sure if is it relevant for docs or not.
I’ve also added a specific error for the case when you run your node as a grantee, but the granter revokes his permission. In this case, your tx will be failed with an error:
`granter has revoked the grant` . This will mean that you have to restart the node w/o a flag.

**Comment by jcstein** [comment]
while testing making feegrant from celestia-appd as we have in the tutorial, using the same accounts as i originally did, it fails when trying to make the same feegrant again: https://mocha.celenium.io/tx/328813ac293516fa43cbdff82e3fee8857c93626a1777cfe1591e3699f75f28a

which is not much of an issue as when i try to increase the amount of the feegrant by a factor of 10, it fails also. meaning, if someone wanted to increase the amount of the feegrant, it may not be possible unless the grant has a balance of 0: https://mocha.celenium.io/tx/4c85d143cd0cf571d242b680b07ebdd9496c417dc6d573ae1b1c7965b419d248

**Comment by jcstein** [comment]
when trying to submit a blob, i first have:

```bash
celestia blob submit 0x42690c204d39600fddd3 0x676d
{
  "result": "rpc error: code = Unknown desc = timed out waiting for tx to be included in a block"
}
```

and then >12 seconds later, submitting again fails with:

```bash
celestia blob submit 0x42690c204d39600fddd3 0x676d --gas.price 0.02
{
  "result": ": incorrect account sequence"
}
```

**Comment by jcstein** [comment]
>  I guess, important to know that such transactions(with granter) will consume more gas than the regular ones.
@vgonkivs is this just due to the transaction size being larger? or something else?

**Comment by jcstein** [comment]
after a few blobs, the balance of celestia10f7nxmcjptwxldjnrrqumfzkxv5l96gnzg2267 is 104 utia

trying to post a blob, from an account with a feegrant (running without the special flag), fails:
```bash
celestia blob submit 0x42690c204d39600fddd3 0x676d
{
  "result": ": insufficient funds"
}
```

I will try this now with celestia-node special flag and only using celestia-node, not combo of celestia-app/node

**Comment by vgonkivs** [comment]
> I guess, important to know that such transactions(with granter) will consume more gas than the regular ones.
@vgonkivs is this just due to the transaction size being larger? or something else?

Yes, because we are specifying the granter via an additional option.

**Comment by jcstein** [comment]
for @vgonkivs vis. i was testing the previous UX and will next test the UX with commands [from his advice above](https://github.com/celestiaorg/docs/pull/1500#issuecomment-2059355222)

**Comment by jcstein** [comment]
1. [feegrant tx from node account 1](https://mocha.celenium.io/tx/b0f4db5d6e96ba4586bfe80e77807653fec79b00259d36deb413b153da42b32e)
2. running with new account with balance 104utia
```bash
celestia light start --p2p.network mocha --core.ip rpc-mocha.pops.one --keyring.accname notjosh --granter.address=celestia1h08dla2jg2529n6vjd7pxsd3gla5dvhptkve53
```
3. submitting a blob from the grantee account is successful https://mocha.celenium.io/tx/374eeffe8cc561bec82fdb7d71d230648756ca74bf9885b5ed18db3438135190
4. balance on grantee account is still 104 utia
5. another test transaction, successful https://mocha.celenium.io/tx/67e7a7d34166c6b17f85d57e1db6c0740617e1470379b80bbe3e0f4c27a0638e

**Comment by jcstein** [comment]
@Bidon15 I plan to remove usage of celestia-app from this PR and just use celestia-node, as it works that way

**Comment by vgonkivs** [comment]
It is also possible to send a fee grant tx to the app directly and then run a node with `--granter.address`

**Comment by vgonkivs** [comment]
The reason why we've added this flag is to allow you at some point stop using granter money(if you want)

**Comment by jcstein** [comment]
awesome. incoming commit with updates to show the new user flow with celestia-node as well as an optional bit on making the allowance tx from celestia-app

can you help me understand what changes UX wise for this:
> 
This PR also added a `gasMultiplier`, for the transactions with the Options.

thank you 🙌

**Comment by jcstein** [comment]
feegrant outputs by celestia-node CLI:

```bash
celestia state grant-fee $GRANTEE_ADDRESS 2000 1000000
{
  "result": "RPC client error: sendRequest failed: http status 200 OK unmarshaling response: EOF"
}
```

and in logs:
```log
2024-04-18T20:42:09.140-0400	INFO	das	das/worker.go:148	sampled header	{"type": "recent", "height": 1639315, "hash": "3814D708A07D4E0D8C79E9746E890CA7BD6A47FE530AA4CEBD7413C5A4EC2B7D", "square width": 64, "data root": "E5614F3603EE4756988131DC4C70D88CFEC8248768AA3AB2AEAB29B4A60E3238", "finished (s)": 0.000005333}
2024-04-18T20:42:12.629-0400	INFO	bitswap-server	server/server.go:547	Bitswap Client ReceiveError: stream reset
2024-04-18T20:42:21.209-0400	INFO	header/store	store/store.go:375	new head	{"height": 1639316, "hash": "71B6A851ECAA30DB380D5F9CF8994DE9CCB279AF669C212CF17E55A3EBED92CC"}
2024-04-18T20:42:21.210-0400	INFO	das	das/worker.go:148	sampled header	{"type": "recent", "height": 1639316, "hash": "71B6A851ECAA30DB380D5F9CF8994DE9CCB279AF669C212CF17E55A3EBED92CC", "square width": 64, "data root": "8193BF804F997D0AC790259F9195B57C72B457FE4DE49EDAFDD870BE9612F7A0", "finished (s)": 0.000019167}
goroutine 24761 [running]:
runtime/debug.Stack()
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/runtime/debug/stack.go:24 +0x64
runtime/debug.PrintStack()
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/runtime/debug/stack.go:16 +0x1c
github.com/cosmos/cosmos-sdk/codec/types.anyCompatError({0x102c073b7, 0xc}, {0x1040908c0, 0x14003d4db60})
	/Users/joshstein/go/pkg/mod/github.com/celestiaorg/cosmos-sdk@v1.20.1-sdk-v0.46.16/codec/types/compat.go:24 +0x3c
github.com/cosmos/cosmos-sdk/codec/types.(*Any).MarshalJSON(0x1056f00f0?)
	/Users/joshstein/go/pkg/mod/github.com/celestiaorg/cosmos-sdk@v1.20.1-sdk-v0.46.16/codec/types/compat.go:56 +0x5c
encoding/json.marshalerEncoder(0x140036d3cc0, {0x1040908c0?, 0x140035f8388?, 0x14004094690?}, {0x78?, 0x0?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:440 +0xc0
encoding/json.structEncoder.encode({{{0x140019d2908, 0xd, 0x10}, 0x14005aeb9e0, 0x14005aeba10}}, 0x140036d3cc0, {0x1040845a0?, 0x140035f8300?, 0x14000846d48?}, {0x0?, ...})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:704 +0x190
encoding/json.ptrEncoder.encode({0x14000846e38?}, 0x140036d3cc0, {0x104015ea0?, 0x140035f8300?, 0x14000846e48?}, {0x20?, 0x39?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:876 +0x1ec
encoding/json.(*encodeState).reflectValue(0x140036d3cc0, {0x104015ea0?, 0x140035f8300?, 0x1012297c0?}, {0xe8?, 0x6e?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:321 +0x70
encoding/json.interfaceEncoder(0x140036d3cc0, {0x103c413e0?, 0x14003c56010?, 0xda00000000000000?}, {0x4?, 0x0?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:658 +0xcc
encoding/json.structEncoder.encode({{{0x140013f8fc8, 0x4, 0x4}, 0x140042286c0, 0x14004228840}}, 0x140036d3cc0, {0x103ec4f80?, 0x14003c56000?, 0x14000847028?}, {0x30?, ...})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:704 +0x190
encoding/json.(*encodeState).reflectValue(0x140036d3cc0, {0x103ec4f80?, 0x14003c56000?, 0x6?}, {0x48?, 0xd?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:321 +0x70
encoding/json.(*encodeState).marshal(0x4?, {0x103ec4f80?, 0x14003c56000?}, {0x80?, 0xff?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:297 +0xc0
encoding/json.(*Encoder).Encode(0x14000847218, {0x103ec4f80, 0x14003c56000})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/stream.go:209 +0xa4
github.com/filecoin-project/go-jsonrpc.(*handler).handle.func1({0x1041007a0?, 0x1400504fde0?})
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/handler.go:488 +0x84
github.com/filecoin-project/go-jsonrpc.withLazyWriter(0x1400617ef30, 0x14000847708)
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/handler.go:507 +0xd0
github.com/filecoin-project/go-jsonrpc.(*handler).handle(0x14003ce9590, {0x104127ff8, 0x14004d86f00}, {{0x14006514ed0, 0x3}, {0x103b98140, 0x14006514ed8}, {0x14006514ee0, 0xe}, {0x14005aec1e0, ...}, ...}, ...)
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/handler.go:487 +0x16d4
github.com/filecoin-project/go-jsonrpc.(*handler).handleReader(0x14003ce9590, {0x104127ff8, 0x14004d86f00}, {0x14d1c3c50, 0x14005b01a40}, {0x1040fdc20, 0x1400383a000}, 0x1040ef2b0)
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/handler.go:268 +0x554
github.com/filecoin-project/go-jsonrpc.(*RPCServer).ServeHTTP(0x1400190a600, {0x104111160, 0x1400383a000}, 0x140036125a0)
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/server.go:104 +0x140
github.com/filecoin-project/go-jsonrpc/auth.(*Handler).ServeHTTP(0x14001568690, {0x104111160, 0x1400383a000}, 0x1400564e360)
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/auth/handler.go:47 +0x378
net/http.serverHandler.ServeHTTP({0x10410aab8?}, {0x104111160?, 0x1400383a000?}, 0x6?)
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/net/http/server.go:3137 +0xbc
net/http.(*conn).serve(0x14004f70000, {0x104127ff8, 0x1400142bce0})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/net/http/server.go:2039 +0x508
created by net/http.(*Server).Serve in goroutine 4996
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/net/http/server.go:3285 +0x3f0
2024-04-18T20:42:24.010-0400	ERROR	rpc	go-jsonrpc@v0.3.1/handler.go:489	json: error calling MarshalJSON for type *types.Any: JSON marshal marshaling error for &Any{TypeUrl:/cosmos.tx.v1beta1.Tx,Value:[10 191 1 10 188 1 10 42 47 99 111 115 109 111 115 46 102 101 101 103 114 97 110 116 46 118 49 98 101 116 97 49 46 77 115 103 71 114 97 110 116 65 108 108 111 119 97 110 99 101 18 141 1 10 47 99 101 108 101 115 116 105 97 49 104 48 56 100 108 97 50 106 103 50 53 50 57 110 54 118 106 100 55 112 120 115 100 51 103 108 97 53 100 118 104 112 116 107 118 101 53 51 18 47 99 101 108 101 115 116 105 97 49 101 53 48 48 108 48 110 108 119 113 106 55 120 53 118 115 113 99 120 113 100 56 114 110 115 53 107 104 118 102 119 48 115 107 103 117 54 48 26 41 10 39 47 99 111 115 109 111 115 46 102 101 101 103 114 97 110 116 46 118 49 98 101 116 97 49 46 66 97 115 105 99 65 108 108 111 119 97 110 99 101 18 31 10 9 18 4 10 2 8 1 24 151 5 18 18 10 12 10 4 117 116 105 97 18 4 50 48 48 48 16 192 132 61 26 64 229 74 212 101 128 129 83 34 15 198 98 219 160 223 112 171 14 51 30 114 245 160 51 100 108 246 37 6 225 70 104 199 62 191 139 119 80 254 139 1 179 221 196 133 58 118 113 13 87 137 18 1 89 213 110 20 196 145 45 148 232 240 214 45],XXX_unrecognized:[]}, this is likely because amino is being used directly (instead of codec.LegacyAmino which is preferred) or UnpackInterfacesMessage is not defined for some type which contains a protobuf Any either directly or via one of its members. To see a stacktrace of where the error is coming from, set the var Debug = true in codec/types/compat.go
2024-04-18T20:42:32.812-0400	INFO	header/store	store/store.go:375	new head	{"height": 1639317, "hash": "B146D4C030B88E8AF4F5433EB1B0C96D38C5AE22D5BA8128C88FCAA9432BED96"}
```

**Comment by jcstein** [comment]
revoking the feegrant also shows an `ERROR` in the logs of celestia-node:

```log
goroutine 56596 [running]:
runtime/debug.Stack()
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/runtime/debug/stack.go:24 +0x64
runtime/debug.PrintStack()
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/runtime/debug/stack.go:16 +0x1c
github.com/cosmos/cosmos-sdk/codec/types.anyCompatError({0x102c073b7, 0xc}, {0x1040908c0, 0x1400761f620})
	/Users/joshstein/go/pkg/mod/github.com/celestiaorg/cosmos-sdk@v1.20.1-sdk-v0.46.16/codec/types/compat.go:24 +0x3c
github.com/cosmos/cosmos-sdk/codec/types.(*Any).MarshalJSON(0x1056f00f0?)
	/Users/joshstein/go/pkg/mod/github.com/celestiaorg/cosmos-sdk@v1.20.1-sdk-v0.46.16/codec/types/compat.go:56 +0x5c
encoding/json.marshalerEncoder(0x140071a3bc0, {0x1040908c0?, 0x14004de2688?, 0x0?}, {0x2?, 0x0?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:440 +0xc0
encoding/json.structEncoder.encode({{{0x140019d2908, 0xd, 0x10}, 0x14005aeb9e0, 0x14005aeba10}}, 0x140071a3bc0, {0x1040845a0?, 0x14004de2600?, 0x5?}, {0x0?, ...})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:704 +0x190
encoding/json.ptrEncoder.encode({0x140013dee38?}, 0x140071a3bc0, {0x104015ea0?, 0x14004de2600?, 0x140013dee48?}, {0x20?, 0x39?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:876 +0x1ec
encoding/json.(*encodeState).reflectValue(0x140071a3bc0, {0x104015ea0?, 0x14004de2600?, 0x1012297c0?}, {0xe8?, 0xee?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:321 +0x70
encoding/json.interfaceEncoder(0x140071a3bc0, {0x103c413e0?, 0x140071a3b90?, 0xda00000100fd7f34?}, {0x4?, 0x0?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:658 +0xcc
encoding/json.structEncoder.encode({{{0x140013f8fc8, 0x4, 0x4}, 0x140042286c0, 0x14004228840}}, 0x140071a3bc0, {0x103ec4f80?, 0x140071a3b80?, 0x140013df078?}, {0xfc?, ...})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:704 +0x190
encoding/json.(*encodeState).reflectValue(0x140071a3bc0, {0x103ec4f80?, 0x140071a3b80?, 0x101226b68?}, {0x38?, 0x0?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:321 +0x70
encoding/json.(*encodeState).marshal(0x105876900?, {0x103ec4f80?, 0x140071a3b80?}, {0x80?, 0xdf?})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/encode.go:297 +0xc0
encoding/json.(*Encoder).Encode(0x140013df218, {0x103ec4f80, 0x140071a3b80})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/encoding/json/stream.go:209 +0xa4
github.com/filecoin-project/go-jsonrpc.(*handler).handle.func1({0x1041007a0?, 0x14007d9c760?})
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/handler.go:488 +0x84
github.com/filecoin-project/go-jsonrpc.withLazyWriter(0x14005174708, 0x140013df708)
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/handler.go:507 +0xd0
github.com/filecoin-project/go-jsonrpc.(*handler).handle(0x14003ce9590, {0x104127ff8, 0x140015cf950}, {{0x14008cb2bf0, 0x3}, {0x103b98140, 0x14008cb2bf8}, {0x140088459f8, 0x14}, {0x140066b64b0, ...}, ...}, ...)
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/handler.go:487 +0x16d4
github.com/filecoin-project/go-jsonrpc.(*handler).handleReader(0x14003ce9590, {0x104127ff8, 0x140015cf950}, {0x14d1c3c50, 0x140044f9480}, {0x1040fdc20, 0x14002f942a0}, 0x1040ef2b0)
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/handler.go:268 +0x554
github.com/filecoin-project/go-jsonrpc.(*RPCServer).ServeHTTP(0x1400190a600, {0x104111160, 0x14002f942a0}, 0x14004ad47e0)
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/server.go:104 +0x140
github.com/filecoin-project/go-jsonrpc/auth.(*Handler).ServeHTTP(0x14001568690, {0x104111160, 0x14002f942a0}, 0x14004ad46c0)
	/Users/joshstein/go/pkg/mod/github.com/filecoin-project/go-jsonrpc@v0.3.1/auth/handler.go:47 +0x378
net/http.serverHandler.ServeHTTP({0x10410aab8?}, {0x104111160?, 0x14002f942a0?}, 0x6?)
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/net/http/server.go:3137 +0xbc
net/http.(*conn).serve(0x14007478090, {0x104127ff8, 0x1400142bce0})
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/net/http/server.go:2039 +0x508
created by net/http.(*Server).Serve in goroutine 4996
	/Users/joshstein/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-arm64/src/net/http/server.go:3285 +0x3f0
2024-04-18T20:52:04.122-0400	ERROR	rpc	go-jsonrpc@v0.3.1/handler.go:489	json: error calling MarshalJSON for type *types.Any: JSON marshal marshaling error for &Any{TypeUrl:/cosmos.tx.v1beta1.Tx,Value:[10 148 1 10 145 1 10 43 47 99 111 115 109 111 115 46 102 101 101 103 114 97 110 116 46 118 49 98 101 116 97 49 46 77 115 103 82 101 118 111 107 101 65 108 108 111 119 97 110 99 101 18 98 10 47 99 101 108 101 115 116 105 97 49 104 48 56 100 108 97 50 106 103 50 53 50 57 110 54 118 106 100 55 112 120 115 100 51 103 108 97 53 100 118 104 112 116 107 118 101 53 51 18 47 99 101 108 101 115 116 105 97 49 101 53 48 48 108 48 110 108 119 113 106 55 120 53 118 115 113 99 120 113 100 56 114 110 115 53 107 104 118 102 119 48 115 107 103 117 54 48 18 31 10 9 18 4 10 2 8 1 24 154 5 18 18 10 12 10 4 117 116 105 97 18 4 50 48 48 48 16 192 132 61 26 64 220 132 51 69 189 210 96 168 75 128 40 142 143 99 99 233 110 198 199 186 212 248 30 52 231 19 187 17 78 58 254 95 108 71 252 239 217 198 172 95 213 184 211 218 216 207 16 95 142 133 156 114 63 222 111 148 171 237 11 85 160 118 62 11],XXX_unrecognized:[]}, this is likely because amino is being used directly (instead of codec.LegacyAmino which is preferred) or UnpackInterfacesMessage is not defined for some type which contains a protobuf Any either directly or via one of its members. To see a stacktrace of where the error is coming from, set the var Debug = true in codec/types/compat.go
```

although [this revoke-fee-grant transaction was also successful](https://mocha.celenium.io/tx/f16cc8d5133322adc12549efbb3f98478770cd2c3b7319e81b195b6e579aeae0), the CLI does not appear to really tell the user this:

```bash
celestia state revoke-grant-fee $GRANTEE_ADDRESS 2000 1000000
{
  "result": "RPC client error: sendRequest failed: http status 200 OK unmarshaling response: EOF"
}
```

**Comment by jcstein** [comment]
| Description                    | TX 1                                                                                               | TX 2                                                                                               |
|--------------------------------|----------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------|
| 0.000176 fee with feegrant      | [Link](https://mocha.celenium.io/tx/82384c8006c6cf73072ffeb160f78c659447dba1757e4a4f6d5e6684935acc61)    | [Link](https://mocha.celenium.io/tx/83fa70a496eaf4fa21da43c88c1f0bf8f9aa6676ec1d47f183fca948ab418f94)  |
| 0.00016 fee without feegrant    | [Link](https://mocha.celenium.io/tx/9e15dcf7e82288bdf0efc06edf92a30eead60d5ed6518a4721fee1bc34613e2c)    | [Link](https://mocha.celenium.io/tx/a670112dee5bc2001b18225587f2cce86c97016a87d33cc1425b755518050348)  |

**Review by jcstein — APPROVED** [review]
this requires a few changes on node i think before this should end up in docs, but it works

**Review by jcstein — APPROVED** [review]
approving ahead of this being live on Mocha next week

**Pull request merged**